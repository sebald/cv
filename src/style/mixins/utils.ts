import { CSSProperties } from 'react';
import { StyledMixin } from '../theme';

export const propOrEmpty = <P extends keyof CSSProperties>(
  name: P
): StyledMixin<Pick<CSSProperties, P>> => (props: Pick<CSSProperties, P>) => {
  const p = props[name];
  return p ? { [name]: p } : {};
};
