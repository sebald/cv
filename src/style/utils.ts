import { theme } from './theme';

const { htmlFontSize } = theme;

export const px = (value: number) => `${value}px`;

export const pxToRem = (pixel: number, base = htmlFontSize) =>
  `${parseFloat((pixel / base).toFixed(5))}rem`;

export const remToPx = (rem: string, base = htmlFontSize) =>
  parseFloat(rem) * base;

export const ensurePx = (value: number | string, base = htmlFontSize) =>
  typeof value === 'string' ? remToPx(value, base) : value;
