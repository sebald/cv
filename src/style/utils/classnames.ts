export const mergeClassNames = (...names: (string | undefined)[]) => ({
  className: names
    .filter(x => typeof x === 'string')
    .join(' ')
    .trim(),
});
