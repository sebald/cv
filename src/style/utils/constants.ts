export const htmlFontSize = 16;

export const ratio = {
  perfectFourth: 1.333,
  perfectFifth: 1.5,
  goldenRatio: 1.618,
  majorSixth: 1.667,
};
