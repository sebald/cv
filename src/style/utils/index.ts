export * from './classnames';
export * from './constants';
export * from './createMixinComponent';
export * from './interpolation';
export * from './unit';
export * from './typography';
