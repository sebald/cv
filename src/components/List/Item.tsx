import { color } from '../../style';
import { Text } from '../Typography';

export const Item = Text.withComponent('li').extend`
  &:before {
    content: '-';
    color: ${color(palette => palette.grey['400'])};
    padding-right: 10px;
  }
`;
