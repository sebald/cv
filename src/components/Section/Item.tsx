import * as React from 'react';

import { padding, styled } from '../../style';
import { Caption, Subheading } from '../Typography';
import { SectionItemProps } from './types';

const ItemWrapper = styled.div`
  ${padding(0, 0, 3)};
`;

export const Item: React.FC<SectionItemProps> = ({
  title,
  caption,
  children,
}) => (
  <ItemWrapper>
    <Subheading>{title}</Subheading>
    {caption && <Caption>{caption}</Caption>}
    {children}
  </ItemWrapper>
);
