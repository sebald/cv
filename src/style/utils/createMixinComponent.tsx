import React from 'react';

import { styled } from '../styled-components';
import { StyledMixin } from '../theme';

import { mergeClassNames } from './classnames';
import { WithClassName } from './types';

/**
 * Idea stolen from:
 * https://github.com/jxnblk/grid-styled/tree/master/styled-space
 */
const StyledChildren: React.FC<WithClassName> = ({ className, children }) => {
  const styledChildren = React.Children.toArray(children).map((child) =>
    React.isValidElement<any>(child)
      ? React.cloneElement(
          child,
          mergeClassNames(child.props.className, className)
        )
      : child
  );

  return <>{styledChildren}</>;
};

export const createMixinComponent = <P extends {}>(
  ...mixins: StyledMixin<P>[]
) => styled(StyledChildren)([] as any, ...mixins);
