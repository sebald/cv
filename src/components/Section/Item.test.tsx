import * as React from 'react';

import { shallowWithTheme } from '../../test.utils';
import { Item } from './Item';

test('render correct', () => {
  const wrapper = shallowWithTheme(<Item title="Title" />);
  expect(wrapper).toMatchSnapshot();
});

test('render with caption', () => {
  const wrapper = shallowWithTheme(<Item title="Title" caption="Caption" />);
  expect(wrapper).toMatchSnapshot();
});

test('render children', () => {
  const wrapper = shallowWithTheme(<Item title="Title">child</Item>);
  expect(wrapper).toMatchSnapshot();
});
