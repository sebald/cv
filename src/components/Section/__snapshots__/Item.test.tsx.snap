// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render children 1`] = `
<styled.div>
  <Subheading>
    Title
  </Subheading>
  child
</styled.div>
`;

exports[`render correct 1`] = `
<styled.div>
  <Subheading>
    Title
  </Subheading>
</styled.div>
`;

exports[`render with caption 1`] = `
<styled.div>
  <Subheading>
    Title
  </Subheading>
  <Caption>
    Caption
  </Caption>
</styled.div>
`;
